# Ville de Québec, Wikidata et livres: une exploration

## À propos

Ce carnet de note s'inscrit dans les recherches du projet [TAMIS](https://tamis.ca/) pour explorer la présence littéraire de la ville de Québec dans Wikidata. Plus de détails au sujet de cette démarche seront publiées sur le carnet de TAMIS.

## Notes techniques

Ce document est un carnet Jupyter Notebook configuré pour faire des requêtes SPARQL en s'inspirant directement des [explications de Bob DuCharme](http://www.bobdc.com/blog/jupytersparql/). Les requêtes sont réalisées directement sur le «endpoint» SPAQL de Wikidata:

In [2]:
%endpoint https://query.wikidata.org/bigdata/namespace/wdq/sparql
%format any
%display table

Endpoint set to: https://query.wikidata.org/bigdata/namespace/wdq/sparql
Request format: ANY
Display: table

## Références

On s'attend à ce que les livres en général soient présents dans Wikidata en respectant les recommandations du projet livres ([anglais](https://www.wikidata.org/wiki/Wikidata:WikiProject_Books), [français](https://www.wikidata.org/wiki/Wikidata:WikiProject_Books/fr)).

Je note au passage quelques numéros d'entités qui seront utilisées fréquemment dans les requêtes qui suivront.

* Livre: Q571
* Version, édition ou traduction: Q3331189
* Maison d'éditon: Q2085381
* Oeuvre littéraire: Q7725634
* Écrivain: Q36180
* Auteur: Q482980
* Poète: Q49757
* Ville de Québec: Q2145
* Quelques villes références: Montréal Q340; Paris Q90; New-York City Q60
* Quelques villes créatives de l'UNESCO en littérature: Angoulème Q134346, Barcelone Q1492, Dublin Q1761, Milan Q490, Seattle Q5083

Je note également quelques propriétés utiles.

* lieu (P276) (endroit où se situe l'objet, la structure ou l'événement. Dans le cas d'une entité administrative utiliser P131. Dans le cas d'une entité géographique utiliser P706.)
* localisation administrative (P131)
* localisation géographique (P706)
* sous-classe de (P279)
* nature de l'élément (P31)
* siège (P159)
* éditeur (P123)
* scolarité (P69)
* lieu de naissance (P19)

## Exploration sur les maisons d'édition

On veut avoir la liste des maisons d'édition de Québec présentes sur Wikidata (à noter l'utilisation de P159 et de P131... P159 devrait suffire, mais on trouve plus de résultats en utilisant aussi P131).

In [3]:
SELECT distinct ?publisher ?publisherLabel
WHERE
{
    ?publisher (wdt:P31/wdt:P279*) wd:Q2085381 .
    ?publisher (wdt:P159*|wdt:P131*) wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

publisher 
 publisherLabel http://www.wikidata.org/entity/Q3402573 
 Presses de l'Université du Québec http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q16684371 
 Q16684371 http://www.wikidata.org/entity/Q2986483 
 Commission de la capitale nationale du Québec http://www.wikidata.org/entity/Q97044787 
 The Chronicle-Telegraph Pub. Co. http://www.wikidata.org/entity/Q95623497 
 Librairie Garneau http://www.wikidata.org/entity/Q3579664 
 Éditions du Septentrion http://www.wikidata.org/entity/Q3579394 
 Le Dauphin blanc http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q2338135 
 Musée national des beaux-arts du Québec http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q16506987 
 Bibliothèque de l'Université Laval http://www.wikidata.org/entity/Q326282 
 Université du Québec http://www.wikidata.org/entity/Q1705530 
 Musée de la civilisation Total: 14, Shown: 14

À noter la présence d'organisations disparues (Dussault & Proulx), qu'on n'aurait pas identifiées comme maisons d'éditions _a priori_ (le Musée de la civilisation ou la CCNQ), ou avec des fiches incomplètes (Q16684371 qui correspond aux Éditions Alto).

Si on s'intéresse aux éditions publiées par des entités de Québec (ici, on ne spécifie pas qu'on cherche des maisons d'édition, parce qu'on part des éditions, donc on assume qu'elles sont publiées par des maisons):

In [20]:
SELECT distinct ?edition ?editionLabel ?publisher ?publisherLabel
WHERE
{
    ?edition (wdt:P31/wdt:P279*) wd:Q3331189 .
    ?edition wdt:P123 ?publisher .
    ?publisher (wdt:P159*/wdt:P131*) wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

edition 
 editionLabel 
 publisher 
 publisherLabel http://www.wikidata.org/entity/Q63615887 
 Dialogue avec l'histoire : les collections de la Bibliothèque de l'Université Laval 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q56816764 
 Le registre de Sillery (1638-1690) 
 http://www.wikidata.org/entity/Q3402573 
 Presses de l'Université du Québec http://www.wikidata.org/entity/Q29018702 
 Le Traité de Paris de 1763 en bref 
 http://www.wikidata.org/entity/Q3579664 
 Éditions du Septentrion http://www.wikidata.org/entity/Q58849509 
 Le registre de Sillery 
 http://www.wikidata.org/entity/Q3402573 
 Presses de l'Université du Québec http://www.wikidata.org/entity/Q63373321 
 Bibliothèque de l'Université Laval : 165 ans d'histoire : 1852-2017 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q51513444 
 Les oisseaux insectivores 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q40881798 
 Allumez vos lampes, s’il vous plaît !!! 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q98073693 
 Le Palais Montcalm : 75 ans d'émotions 
 http://www.wikidata.org/entity/Q2986483 
 Commission de la capitale nationale du Québec http://www.wikidata.org/entity/Q98073693 
 Le Palais Montcalm : 75 ans d'émotions 
 http://www.wikidata.org/entity/Q98073919 
 Société du Palais Montcalm http://www.wikidata.org/entity/Q51502553 
 Les mauvaises herbes dans la province de Québec et différents moyens de les détruire 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q54165640 
 Q54165640 
 http://www.wikidata.org/entity/Q3579394 
 Le Dauphin blanc http://www.wikidata.org/entity/Q58341258 
 Statistiques sociales 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q77009625 
 Dictionnaire des artistes de langue française en Amérique du Nord 
 http://www.wikidata.org/entity/Q2338135 
 Musée national des beaux-arts du Québec http://www.wikidata.org/entity/Q78880939 
 Q78880939 
 http://www.wikidata.org/entity/Q3402573 
 Presses de l'Université du Québec http://www.wikidata.org/entity/Q52182553 
 Q52182553 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q91949031 
 Q91949031 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q77009625 
 Dictionnaire des artistes de langue française en Amérique du Nord 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q51513428 
 Les oiseaux de la province de Québec 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q51468513 
 L'élevage du cheval en Canada 
 http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx Total: 19, Shown: 19

## Exploration sur les maisons d'éditions dans d'autres villes

Pour se donner des points de repère, on veut comparer avec les autres villes créatives l'UNESCO en littérature, et nos villes de référence.

In [6]:
SELECT distinct ?publisher ?publisherLabel ?city ?cityLabel
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?publisher (wdt:P31/wdt:P279*) wd:Q2085381 .
    ?publisher (wdt:P159*|wdt:P131*) ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

publisher 
 publisherLabel 
 city 
 cityLabel http://www.wikidata.org/entity/Q633812 
 Archie Comics 
 http://www.wikidata.org/entity/Q60 
 New York City http://www.wikidata.org/entity/Q2068658 
 505 Games 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q380568 
 Feltrinelli 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q64814857 
 Voodoo 
 http://www.wikidata.org/entity/Q90 
 Paris http://www.wikidata.org/entity/Q77369313 
 Stumbling Cat 
 http://www.wikidata.org/entity/Q5083 
 Seattle http://www.wikidata.org/entity/Q3579355 
 Hurtubise 
 http://www.wikidata.org/entity/Q340 
 Montreal http://www.wikidata.org/entity/Q2075464 
 Dell Publishing 
 http://www.wikidata.org/entity/Q60 
 New York City http://www.wikidata.org/entity/Q8965148 
 Grup 62 
 http://www.wikidata.org/entity/Q1492 
 Barcelona http://www.wikidata.org/entity/Q23301680 
 Kitfox Games 
 http://www.wikidata.org/entity/Q340 
 Montreal http://www.wikidata.org/entity/Q273835 
 Éditions Larousse 
 http://www.wikidata.org/entity/Q90 
 Paris http://www.wikidata.org/entity/Q2080179 
 Columbia University Press 
 http://www.wikidata.org/entity/Q60 
 New York City http://www.wikidata.org/entity/Q79449727 
 Éditions de Trévise 
 http://www.wikidata.org/entity/Q90 
 Paris http://www.wikidata.org/entity/Q2874472 
 Avon 
 http://www.wikidata.org/entity/Q60 
 New York City http://www.wikidata.org/entity/Q6801272 
 McGill-Queen's University Press 
 http://www.wikidata.org/entity/Q340 
 Montreal http://www.wikidata.org/entity/Q60616868 
 The Family Collective 
 http://www.wikidata.org/entity/Q60 
 New York City http://www.wikidata.org/entity/Q355301 
 Adelphi Edizioni 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q1225116 
 Les Belles Lettres 
 http://www.wikidata.org/entity/Q90 
 Paris http://www.wikidata.org/entity/Q684254 
 Drawn & Quarterly 
 http://www.wikidata.org/entity/Q340 
 Montreal http://www.wikidata.org/entity/Q54117155 
 Adage Games 
 http://www.wikidata.org/entity/Q340 
 Montreal http://www.wikidata.org/entity/Q1892146 
 Presses Universitaires de France 
 http://www.wikidata.org/entity/Q90 
 Paris Total: 1033, Shown: 20

Avec 1033 maisons d'édition dans les 9 villes échantillonnées, on a une moyenne de 114 maisons par ville (8 fois plus qu'à Québec). Si on veut faire le décompte par ville:

In [8]:
SELECT ?cityLabel (count(distinct ?publisher) as ?pubCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?publisher (wdt:P31/wdt:P279*) wd:Q2085381 .
    ?publisher (wdt:P159*|wdt:P131*) ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 pubCount Québec 
 14 Barcelone 
 136 Seattle 
 22 Milan 
 162 Montréal 
 46 Paris 
 369 New York 
 265 Dublin 
 18 Angoulême 
 1 Total: 9, Shown: 9

Évidemment, ces données reflètent _les maisons d'édition dont la fiche Wikidata mentionne la ville_, dans la propriété P159 ou P131.

## Exploration sur les auteurs

Les exemples de Wikidata suggèrent de chercher des instances d'écrivain, d'auteurs ou de poètes pour trouver des auteurs de façon générale. Je tente d'abord de trouver des auteurs qui ont une relation _directe_ (sans passer par d'autres entités), peu importe le type, avec Québec:

In [11]:
SELECT distinct ?writer ?writerLabel ?relation ?relationLabel
WHERE
{
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel wd:Q2145
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}

writer 
 writerLabel 
 relation 
 relationLabel http://www.wikidata.org/entity/Q2836600 
 Alice Lemieux-Lévesque 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3092329 
 Félix-Antoine Savard 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q660646 
 Pierre-Joseph-Olivier Chauveau 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q6141049 
 James Pattison Cockburn 
 http://www.wikidata.org/entity/P937 
 lieu de travail http://www.wikidata.org/entity/Q3275641 
 Madeleine Ferron 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3320823 
 Monique Corriveau 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q2967448 
 Chrystine Brouillet 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q2834952 
 Alfred Bailey 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q2425095 
 Philippe Sylvain 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3100343 
 Roland-Michel Tremblay 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3058987 
 Esther Rochon 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3103738 
 Georgette Lacroix 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3158713 
 Jacques Desautels 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3184265 
 Joseph-François Perrault 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3336057 
 Narcisse-Henri-Édouard Faucher de Saint-Maurice 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3188043 
 Judy Quinn 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3189273 
 Julie Stanton 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3169374 
 Jean-Pierre Guay 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3291644 
 Marie-Christine Lê-Huu 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q2958277 
 Charles Baillairgé 
 http://www.wikidata.org/entity/P20 
 lieu de mort Total: 209, Shown: 20

On trouve 209 références. Refaisons l'exercice avec nos villes exemples pour comparer:

In [14]:
SELECT distinct ?writer ?writerLabel ?relation ?relationLabel ?city ?cityLabel
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}

writer 
 writerLabel 
 relation 
 relationLabel 
 city 
 cityLabel http://www.wikidata.org/entity/Q253513 
 Ally Sheedy 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q435135 
 Mary Rodgers 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q120432 
 Jane Bowles 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q113105 
 James Gleick 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q47899 
 Paris Hilton 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q438164 
 Samuel R. Delany 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q240959 
 Emma Lazarus 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q79904 
 J. D. Salinger 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q108283 
 Alan Arkin 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q465594 
 Joy Davidman 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q468128 
 Theodore Nicholas Gill 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q149489 
 Jacob M. Appel 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q236950 
 Erica Jong 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q380784 
 James Siegel 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q155649 
 Richard Holbrooke 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q433134 
 Judith Lewis Herman 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q234255 
 Cecily von Ziegesar 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q471337 
 Ed Lacy 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q444125 
 Diana Barrymore 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q32931 
 Sarah Kay 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York Total: 15740, Shown: 20

Et si on veut un décompte du nombre d'auteurs ayant des relations avec les villes en question, par ville:

_(cette requête prend trop de temps à l'exécuter, il n'est pas possible d'obtenir un résultat pour l'instant)_

In [20]:
SELECT ?cityLabel (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?cityLabel

Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT ?cityLabel (count(distinct ?writer) as ?writerCount)\nWHERE\n{\n  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }\n  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }\n  ?writer ?rel ?city\n  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n  ?relation wikibase:directClaim ?rel .\n}\nGROUP BY ?cityLabel\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\n\tat org.e

Et on veut connaître les types de relations les plus fréquents.

In [18]:
SELECT ?relation (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?relation


relation 
 writerCount http://www.wikidata.org/entity/P69 
 2 http://www.wikidata.org/entity/P19 
 7322 http://www.wikidata.org/entity/P108 
 2 http://www.wikidata.org/entity/P119 
 16 http://www.wikidata.org/entity/P101 
 1 http://www.wikidata.org/entity/P27 
 1 http://www.wikidata.org/entity/P551 
 711 http://www.wikidata.org/entity/P276 
 5 http://www.wikidata.org/entity/P740 
 3 http://www.wikidata.org/entity/P937 
 1149 http://www.wikidata.org/entity/P20 
 6472 Total: 11, Shown: 11

Les relations les plus fréquentes sont donc le lieu de naissance (P19), de décès (P20), de résidence (P551) et de travail (P937).

La ville de scolarité semble très peu documentée, mais on peut émettre l'hypothèse que si on passe par l'institution d'enseignement, on aura plus de résultats. Par exemple, pour Québec:


In [22]:
SELECT distinct ?writer ?writerLabel ?school ?schoolLabel
WHERE
{
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer wdt:P69 ?school .
  ?school (wdt:P159*/wdt:P131*) wd:Q2145
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}


writer 
 writerLabel 
 school 
 schoolLabel http://www.wikidata.org/entity/Q2851659 
 Annie Groovie 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q298358 
 Marie-Claire Blais 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3380487 
 Philippe Panneton 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3291628 
 Marie-Christine Bernard 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3496855 
 Stanley Péan 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q16663531 
 Martin Blais 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3301025 
 Maurice Lebel 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q1468786 
 Nicolas Dickner 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q57084146 
 Jacques Mathieu 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q1150878 
 Noël Simard 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3165792 
 Jean-François Nadeau 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3260781 
 Louis-Philippe Audet 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3126438 
 Hamidou Dia 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q58043349 
 Thérèse Laferrière 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3375589 
 Perrine Leblanc 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3262812 
 Louis O'Neill 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q2872028 
 Aurélien Boivin 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3426988 
 René Vézina 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q56857544 
 Richard Leclerc 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q2870697 
 Aude 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval Total: 186, Shown: 20

## Vrac / à explorer

Je vois Nicolas Dickner dans les auteurs qui ont étudié à Québec, à l'Université Laval. Sa fiche contient des oeuvres notables, Tarmac et Nikolski. Les deux fiches portent sur les oeuvres et ne contiennent pas de référence aux éditions (et donc pas aux ISBN). Aucune fiche ne fait de lien avec les Éditions Alto.